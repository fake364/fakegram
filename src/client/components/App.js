import React, {Component} from 'react';
import Header from "./mainapp/Header";
import {Redirect, Route, Switch} from "react-router-dom";
import User from "./mainapp/User";
import Auth from "./auth/Auth";
import Main from "./mainapp/Main";
import LoginComponent from "./auth/Login";
import {connect} from "react-redux";
import axios from "axios";
import CircularProgress from "@material-ui/core/CircularProgress";
import PostCreator from "./mainapp/PostCreator";

class App extends Component {

    componentDidMount() {
        this.props.checkToken();
    }

    render() {
        const MyProgress = () => {
            return <main>
                <div className="MainFlex">
                    <CircularProgress/>
                </div>
            </main>
        }
        return (
            <div className="App">
                {this.props.isLogined ? <Header/> : null}
                <Switch>
                    <Route exact path="/">
                        {this.props.isLogined ? <Main/> : this.props.isLoading ? <MyProgress/> : <Auth/>}
                    </Route>
                    <Route exact path="/login">
                        {this.props.isLogined ? <Redirect to="/"/> : null}
                        {this.props.isLogined ? <Main/> : this.props.isLoading ? <MyProgress/> : <LoginComponent/>}
                    </Route>
                    <Route exact path="/post">
                        {this.props.isLogined ? <PostCreator/> : this.props.isLoading ? <MyProgress/> :
                            <LoginComponent/>}
                    </Route>
                    <Route exact path="/:user" component={User}/>

                </Switch>
            </div>

        );
    }
}

export default connect(state => ({
    isLogined: state.authReducer.isLogined,
    isLoading: state.authReducer.isLoading
}), dispatch => ({
    checkToken: () => {
        const asyncCheck = () => dispatch => {
            dispatch({type: "LOGIN_INIT"})
            axios.get('api/checkToken')
                .then(res => {
                    if (res.status === 200) {
                        dispatch({type: "LOGIN_SUCCEED"});
                        dispatch({
                            type: "USER_INFO_GOTTEN",
                            payload: {username: res.data.username, userid: res.data.userid, name: res.data.name}
                        });
                        console.log(res.data);
                    } else {
                        const error = new Error(res.error);
                        throw error;
                    }
                })
                .catch(err => {
                    dispatch({type: "LOGIN_FAILED"});
                });

        }
        dispatch(asyncCheck());
    },

}))(App);
