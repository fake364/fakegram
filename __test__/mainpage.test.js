import {describe, expect, it} from "@jest/globals";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import React from "react";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import {Router} from "react-router-dom";
import App from "../src/client/components/App";
import {createMemoryHistory} from "history";
import initState from "./__mocks__/user";

describe("Main page", () => {

    it("should render correctly if im not authed", () => {
        let elementTree;
        const mockStore = configureStore([thunk]);
        let store = mockStore(initState.userNotFoundAndNotAuthed);
        const history = createMemoryHistory();
        history.push('/');
        elementTree = renderer.create(<Provider
            store={store}><Router history={history}><App/></Router></Provider>);
        expect(elementTree.toJSON()).toMatchSnapshot();
    });

    it("should render correctly if im authed", () => {
        let elementTree;
        const mockStore = configureStore([thunk]);
        let store = mockStore(initState.userNotFoundAndAuthed);
        const history = createMemoryHistory();
        history.push('/');
        elementTree = renderer.create(<Provider
            store={store}><Router history={history}><App/></Router></Provider>);
        expect(elementTree.toJSON()).toMatchSnapshot();
    });
});