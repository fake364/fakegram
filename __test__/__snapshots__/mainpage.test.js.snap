// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page should render correctly if im authed 1`] = `
<div
  className="App"
>
  <div
    className="Header"
  >
    <nav>
      <div
        className="SubHead"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="logotitle"
          >
            <div>
              <img
                alt="text"
                src="/dist/build/images/logo192.png"
              />
            </div>
            <div
              className="leftborder"
            >
              Fakegram
            </div>
          </div>
        </a>
        <div>
          <input
            placeholder="Поиск"
            type="text"
          />
        </div>
        <div
          className="Panel"
        >
          <img
            alt="Photo2"
            className="Icon"
            src="/dist/build/images/heart.png"
          />
          <a
            href="/fake"
            onClick={[Function]}
          >
            <img
              alt="Photo3"
              className="Icon"
              src="/dist/build/images/user.png"
            />
          </a>
          <a
            href="/post"
            onClick={[Function]}
          >
            <img
              alt="Photo3"
              className="Icon"
              src="/dist/build/images/upload.png"
            />
          </a>
          <button
            className="MuiButtonBase-root MuiIconButton-root makeStyles-root-1"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <img
                src="/dist/build/images/logout.png"
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
            </span>
          </button>
        </div>
      </div>
    </nav>
  </div>
  <main>
    <div
      className="MainFlex"
    >
      <div
        className="PostsFeed"
      >
        <p>
          Лента постов пуста! Подпишитесь на других пользователей или создайте пост
        </p>
      </div>
      <div
        className="Stories"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <img
            alt="Photo"
            className="Avatar"
            src="/dist/build/images/avatar.jpg"
          />
          <div
            style={
              Object {
                "marginLeft": "14px",
              }
            }
          >
            <h2>
              fake
            </h2>
            <p>
              Valentin Serebreanu
            </p>
          </div>
        </div>
        <div
          className="StoriesList"
        >
          <div
            className="StoriesHeader"
          >
            <p>
              Истории
            </p>
            <h3>
              Смотреть все
            </h3>
          </div>
          <div
            className="ScrollList"
          >
            <div
              className="Story"
            >
              <img
                alt="Photo"
                src="/dist/build/images/avatar.jpg"
              />
              <div
                className="nickmar"
              >
                <h2>
                  bolkonski_v
                </h2>
                <p>
                  1 ЧАС НАЗАД
                </p>
              </div>
            </div>
            <div
              className="Story"
            >
              <img
                alt="Photo"
                src="/dist/build/images/avatar.jpg"
              />
              <div
                className="nickmar"
              >
                <h2>
                  bolkonski_v
                </h2>
                <p>
                  1 ЧАС НАЗАД
                </p>
              </div>
            </div>
            <div
              className="Story"
            >
              <img
                alt="Photo"
                src="/dist/build/images/avatar.jpg"
              />
              <div
                className="nickmar"
              >
                <h2>
                  bolkonski_v
                </h2>
                <p>
                  1 ЧАС НАЗАД
                </p>
              </div>
            </div>
            <div
              className="Story"
            >
              <img
                alt="Photo"
                src="/dist/build/images/avatar.jpg"
              />
              <div
                className="nickmar"
              >
                <h2>
                  bolkonski_v
                </h2>
                <p>
                  1 ЧАС НАЗАД
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Main page should render correctly if im not authed 1`] = `
<div
  className="App"
>
  <main>
    <div
      className="MainFlex"
    >
      <div
        className="Auth"
      >
        <div
          className="Phones"
        >
          <div
            className="Slides"
          >
            <img
              src="/dist/build/images/first.jpg"
            />
          </div>
        </div>
        <div
          className="SecondSlide"
        >
          <div
            className="RegisterForm"
          >
            <h1>
              <i>
                Fakegram
              </i>
            </h1>
            <p>
              Зарегистрируйтесь, чтобы смотреть фото и видео ваших друзей.
            </p>
            <form
              onSubmit={[Function]}
            >
              <input
                name="phone"
                onChange={[Function]}
                placeholder="Моб телефон"
                required={true}
                type="text"
              />
              <input
                name="name"
                onChange={[Function]}
                placeholder="Имя и фамилия"
                required={true}
                type="text"
              />
              <input
                name="login"
                onChange={[Function]}
                placeholder="Имя пользователя"
                required={true}
                type="text"
              />
              <input
                name="password"
                onChange={[Function]}
                placeholder="Пароль"
                required={true}
                type="password"
              />
              <button>
                Регистрация
              </button>
            </form>
            <p
              style={
                Object {
                  "fontSize": "14px",
                  "padding": "8px 24px",
                }
              }
            >
              Регистрируясь, вы принимаете наши Условия, Политику использования данных и Политику в отношении файлов cookie
            </p>
          </div>
          <div
            className="IfHasLogin"
          >
            Есть аккаунт? 
            <a
              href="/login"
              onClick={[Function]}
            >
              Вход
            </a>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
